pkg/okteto/pipeline.go

```diff
@@ -17,24 +17,28 @@ import (
 	"context"
 	"fmt"
 	"strings"
+	"time"
 
 	"github.com/okteto/okteto/pkg/config"
 	oktetoErrors "github.com/okteto/okteto/pkg/errors"
 	oktetoLog "github.com/okteto/okteto/pkg/log"
+	oktetoTime "github.com/okteto/okteto/pkg/time"
 	"github.com/okteto/okteto/pkg/types"
 	"github.com/shurcooL/graphql"
 	giturls "github.com/whilp/git-urls"
 )
 
 type pipelineClient struct {
-	client *graphql.Client
-	url    string
+	client                    graphqlClientInterface
+	url                       string
+	tickerWithTimeoutProvider func(ticker, timeout time.Duration) oktetoTime.TickerWithTimeoutInterface
```

Comment:
I feel that the `time` package is not something we'll like long term. It's a very thin layer of wrapping that could exparse to the whole app without much gains and limiting ourselves to use golang's native concurrency model

I feel that this function itself it's a great way of handling the depnedency injection. I'd just use 2 functions here that would provide the 2 tickers

You can easily mock tickers by creating the struct with the channel, here you have an example https://github.com/okteto/app/blob/700b10e0e264beb2f3357ee724fcd027a12b5aa6/backend/pkg/controlplane/federatednamespace/crud/create_test.go#L33. This means that maybe there's no need for a provider at all, and you could directly have the tickers

Reply:
Interesting discussion, I agree that it would be nicer to remove the abstraction/wrapper around the native `time` and use dependency injection as much as possible. I haven't tried refactoring this PR using that approach so I am not sure it's worth it or it's better to keep the provider approach.

In a much simpler scenario, I've written a function that allows injecting both `ticker` and `timeout` so it's not hard to mock in testing: https://github.com/okteto/okteto/pull/3455#discussion_r1142796151

Having said this, I am not against merging this which is already an improvement. We can always iterate on this later, if we want.

Reply:
I get what you say but I need to have the timer for the timeout and the ticker sync between them to know how many times should I call the ticker before calling the timeout function. I think all this logic  can be refactored into a smaller approach. Let me think about it and will come back with a new approach!!

Reply:
Done at [9cbd78b](https://github.com/okteto/okteto/pull/3458/commits/9cbd78bdf749d871e97d6afc1b5f8f0251633981)